src/main/java/ai/knowly/langtorch/processor/cohere/CohereProcessorModule.java

```diff
@@ -0,0 +1,24 @@
+package ai.knowly.langtorch.processor.cohere;
+
+import ai.knowly.langtorch.llm.cohere.CohereApiService;
+import ai.knowly.langtorch.utils.Environment;
+import ai.knowly.langtorch.utils.api.key.CohereKeyUtil;
+import com.google.common.flogger.FluentLogger;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+
+import java.time.Duration;
+
+import static ai.knowly.langtorch.llm.cohere.CohereApiService.buildApi;
+
+public final class CohereProcessorModule extends AbstractModule {
```

Comment:
I was originally using the Guice module too but I was also wondering later on: does this mean we have to enforce our users to use Guice as the dependency injection framework?
but it's okay if it's only for testing.

Reply:
yea, we should go with Guice.
https://github.com/Knowly-ai/langtorch/issues/119